# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/GtkCompositorWidget.cpp
# Commit: d8a0164db5db
# Full Hash: d8a0164db5db7090fd7549b03dd0391f7151649c
# Author: stransky <stransky@redhat.com>
# Date: 2024-07-13 09:15:14
# Description:
#   Bug 1907511 [Linux/X11] Check mWindow at GtkCompositorWidget::LockSurface() r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D216380
# ==============================================================================

diff -r e243955016ff -r d8a0164db5db widget/gtk/GtkCompositorWidget.cpp
--- a/widget/gtk/GtkCompositorWidget.cpp	Fri Jul 12 21:57:33 2024 +0300
+++ b/widget/gtk/GtkCompositorWidget.cpp	Fri Jul 12 18:12:51 2024 +0000
@@ -212,7 +212,7 @@
 #endif
 
 UniquePtr<MozContainerSurfaceLock> GtkCompositorWidget::LockSurface() {
-  return mWidget->LockSurface();
+  return mWidget ? mWidget->LockSurface() : nullptr;
 }
 
 }  // namespace widget
